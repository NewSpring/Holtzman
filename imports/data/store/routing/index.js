
// import "./saga"

// Constants
// taken from https://github.com/reactjs/react-router-redux/blob/master/src/index.js
const TRANSITION = "@@router/TRANSITION";
const UPDATE_LOCATION = "@@router/UPDATE_LOCATION";

const SELECT_LOCATION = (state) => state.routing.location;

function transition(method) {
  return (...args) => ({
    type: TRANSITION,
    payload: { method, args },
  });
}

export const push = transition("push");
export const replace = transition("replace");
export const go = transition("go");
export const goBack = transition("goBack");
export const goForward = transition("goForward");

export const routeActions = { push, replace, go, goBack, goForward };

function updateLocation(location) {
  return {
    type: UPDATE_LOCATION,
    payload: location,
  };
}

// Reducer

const initialState = {
  location: undefined,
};

export function routeReducer(state = initialState, { type, payload: location }) {
  if (type !== UPDATE_LOCATION) {
    return state;
  }

  return { ...state, location };
}

// Syncing

export function syncHistory(history) {
  let unsubscribeHistory;
  let currentKey;
  let unsubscribeStore;

  let connected = false;
  let syncing = false;

  history.listen((location) => { initialState.location = location; })();

  function middleware(store) {
    unsubscribeHistory = history.listen((location) => {
      currentKey = location.key;
      if (syncing) {
        // Don't dispatch a new action if we're replaying location.
        return;
      }
      const { routing } = store.getState();
      if (!location.previous) {
        location.previous = []; // eslint-disable-line no-param-reassign
      }
      if (!routing.location.previous) {
        routing.location.previous = [];
      }

      if (routing.location.previous[routing.location.previous.length - 1] === location.pathname) {
        routing.location.previous.splice(-1);
        location.previous = routing.location.previous; // eslint-disable-line no-param-reassign
      } else {
        // eslint-disable-next-line no-param-reassign
        location.previous = [...routing.location.previous, ...[routing.location.pathname]];
      }


      store.dispatch(updateLocation(location));
    });

    connected = true;

    // eslint-disable-next-line consistent-return
    return (next) => (action) => {
      if (action.type !== TRANSITION || !connected) {
        return next(action);
      }

      const { payload: { method, args } } = action;

      history[method](...args);
    };
  }

  middleware.listenForReplays =
    (store, selectLocationState = SELECT_LOCATION) => {
      const getLocationState = () => selectLocationState(store.getState());
      const initialLocation = getLocationState();

      unsubscribeStore = store.subscribe(() => {
        const location = getLocationState();

        // If we're resetting to the beginning, use the saved initial value. We
        // need to dispatch a new action at this point to populate the store
        // appropriately.
        if (location.key === initialLocation.key) {
          history.replace(initialLocation);
          return;
        }

        // Otherwise, if we need to update the history location, do so without
        // dispatching a new action, as we're just bringing history in sync
        // with the store.
        if (location.key !== currentKey) {
          syncing = true;
          history.transitionTo(location);
          syncing = false;
        }
      });
    };

  middleware.unsubscribe = () => {
    unsubscribeHistory();
    if (unsubscribeStore) {
      unsubscribeStore();
    }

    connected = false;
  };

  return middleware;
}
