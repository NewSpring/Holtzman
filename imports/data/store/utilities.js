
// stored state for use with other packages
const reducers = {};

const addReducer = obj => {
  // eslint-disable-next-line no-restricted-syntax, guard-for-in
  for (const name in obj) {
    const handler = obj[name];

    if (reducers[name]) {
      throw new Error(
        "Reducer assigned",
        `reducers function ${name} is already registered`
      );
    }

    if (!handler || typeof (handler) !== "function") {
      throw new Error(
        "Reducer TypeError",
        `Reducer ${name} requires a function`
      );
    }

    reducers[name] = handler;
  }

  return obj;
};

const createReducer = (initialState, handlers) => (
  (state = initialState, action) => {
    // better than switch statement
    // eslint-disable-next-line no-prototype-builtins
    if (handlers.hasOwnProperty(action.type)) {
      return handlers[action.type](state, action);
    }
    return state;
  }
);

// stored middlewares for use with other packages
const middlewares = [];

const addMiddleware = (...newWares) => {
  for (const middleware of newWares) { middlewares.push(middleware); }
};

// stored sagas for use with other packages
const sagas = [];

const addSaga = (...newSagas) => {
  for (const saga of newSagas) { sagas.push(() => saga); }
};


export {
  addMiddleware,
  middlewares,

  addReducer,
  createReducer,
  reducers,

  sagas,
  addSaga,
};
