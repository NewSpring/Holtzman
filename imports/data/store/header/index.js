/*
  Share action types
*/
import reducer from "./reducer";
import "./saga";
import { addReducer } from "../utilities";

addReducer({
  header: reducer,
});

export default {
  show: (options = {}) => ({
    type: "HEADER.TOGGLE_VISIBILITY",
    visible: typeof options.visible === "undefined" ? true : options.visible,
    statusBar: options.statusBar,
  }),
  hide: (options = {}) => ({
    type: "HEADER.TOGGLE_VISIBILITY",
    visible: typeof options.visible === "undefined" ? false : options.visible,
    statusBar: options.statusBar,
  }),

  set: (content, requestee) => ({ type: "HEADER.SET", content, requestee }),
  color: (color) => ({ type: "HEADER.SET", content: { color } }),
  title: (title) => ({ type: "HEADER.SET", content: { title } }),
  light: (light) => ({ type: "HEADER.SET", content: { light } }),
  isSearch: (isSearch, searchSubmit) => ({
    type: "HEADER.SET",
    content: { isSearch, searchSubmit },
  }),

  lock: (requestee) => ({ type: "HEADER.LOCK", requestee }),
  unlock: () => ({ type: "HEADER.UNLOCK" }),

  // XXX this currently doesn't affect the store
  statusBarColor: (color) => ({ type: "STATUSBAR.SET", color }),
};
