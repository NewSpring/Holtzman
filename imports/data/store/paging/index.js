import reducer from "./reducer";
import { addReducer } from "../utilities";

addReducer({
  paging: reducer,
});

export default {
  increment: () => ({ type: "PAGING.INCREMENT" }),
  pause: () => ({ type: "PAGING.PAUSE" }),
  resume: () => ({ type: "PAGING.RESUME" }),
  reset: () => ({ type: "PAGING.RESET" }),
};
