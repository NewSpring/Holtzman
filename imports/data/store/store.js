import "regenerator-runtime/runtime";

import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import { ApolloProvider } from "react-apollo";
import createSagaMiddleware from "redux-saga";
import reduxReset from "redux-reset";

import GraphQL from "../graphql";

import { reducers, middlewares, sagas } from "./utilities";
import { syncHistory, routeReducer } from "./routing";

const createReduxStore = (initialState, history) => {
  if (initialState) {
    // bug with SSR
    delete initialState.nav; // eslint-disable-line no-param-reassign
  }

  const joinedReducers = { ...reducers,
    ...{
      routing: routeReducer,
      apollo: GraphQL.reducer(),
    },
  };

  const sharedMiddlewares = [...middlewares, GraphQL.middleware()];

  const reduxRouterMiddleware = syncHistory(history);

  const sagaMiddleware = createSagaMiddleware();

  let sharedCompose = [
    applyMiddleware(
      ...sharedMiddlewares,
      sagaMiddleware,
      reduxRouterMiddleware,
    ),
    reduxReset(),
  ];

  if (process.env.NODE_ENV !== "production") {
    sharedCompose = [...sharedCompose, ...[
      typeof window === "object" && typeof window.devToolsExtension !== "undefined" ?
        window.devToolsExtension() :
        f => f,
    ]];
  }

  const store = compose(...sharedCompose)(createStore)(
    combineReducers(joinedReducers), initialState,
  );

  sagas.forEach(saga => sagaMiddleware.run(saga()));

  return store;
};


const wrapper = ApolloProvider;

export {
  wrapper,
  createReduxStore,
};
