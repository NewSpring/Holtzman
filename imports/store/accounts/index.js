
import { addReducer, addMiddleware } from "../utilities";

import actions from "./actions";
import reducer from "./reducer";
import "./saga";

addReducer({
  accounts: reducer
});

export default actions;
