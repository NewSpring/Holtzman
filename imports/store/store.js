import "regenerator-runtime/runtime";

import { Component, PropTypes } from "react";
import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import { ApolloProvider } from "react-apollo";
import createSagaMiddleware from "redux-saga";
import reduxReset from "redux-reset";

import { GraphQL } from "../graphql";

import { reducers, middlewares, sagas } from "./utilities";
import { syncHistory, routeReducer } from "./routing";

const createReduxStore = (initialState, history) => {

  if (initialState) {
    // bug with SSR
    delete initialState.nav;
  }

  const joinedReducers = {...reducers, ...{
    routing: routeReducer,
    apollo: GraphQL.reducer(),
  }};

  let sharedMiddlewares = [...middlewares, ...GraphQL.middleware()];

  const reduxRouterMiddleware = syncHistory(history);

  const sagaMiddleware = createSagaMiddleware();
  let sharedCompose = [
    applyMiddleware(
      ...sharedMiddlewares,
      sagaMiddleware,
      reduxRouterMiddleware
    ),
    reduxReset(),
  ];

  if (process.env.NODE_ENV != "production") {
    sharedCompose = [...sharedCompose, ...[
      typeof window === "object" && typeof window.devToolsExtension !== "undefined" ? window.devToolsExtension() : f => f
    ]];
  }

  const store = compose(...sharedCompose)(createStore)(
    combineReducers(joinedReducers), initialState
  );

  sagas.forEach(saga => sagaMiddleware.run(saga()));

  return store;

};


const wrapper = ApolloProvider;

export {
  wrapper,
  createReduxStore
};
