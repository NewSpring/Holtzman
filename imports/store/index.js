
import accounts from "./accounts";
import audio from "./audio";
import give from "./give";
import liked from "./liked";
import live from "./live";
import modal from "./modal";
import nav from "./nav";
import paging from "./paging";
import responsive from "./responsive";
import routing from "./routing";
import search from "./search";
import sections from "./sections";
import share from "./share";
import topics from "./topics";
// import fullscreen from "./fullscreen";

let header;
if (process.env.NATIVE) header = require("./header").default;

import { wrapper, createReduxStore } from "./store";
import { addMiddleware, addReducer, createReducer, addSaga } from "./utilities";

export {
  accounts,
  audio,
  give,
  header,
  liked,
  live,
  modal,
  nav,
  paging,
  responsive,
  routing,
  search,
  sections,
  share,
  topics,

  wrapper,
  createReduxStore,

  addMiddleware,

  addReducer,
  createReducer,
};
