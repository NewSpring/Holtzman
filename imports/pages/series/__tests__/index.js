import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { nav as navActions } from "../../../store";
import {
  TemplateWithoutData as Template,
  SERIES_QUERY,
} from "../";

jest.mock("../../../mixins/mixins.Header", () => {});
jest.mock("../../../mixins/mixins.Likeable", () => {});
jest.mock("../../../store/header", () => {});
jest.mock("../../../store", () => ({
  nav: {
    setLevel: jest.fn(),
  },
}));
const defaultProps = {
  dispatch: jest.fn(),
  data: {
    refetch: jest.fn(),
    content: [{}, {}],
  },
  Loading: jest.fn(),
};

const generateComponent = (additionalProps = {}) => {
  const newProps = {
    ...defaultProps,
    ...additionalProps,
  };
  return <Template { ...newProps } />;
};

it("renders with props", () => {
  const wrapper = shallow(generateComponent());
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("renders loading", () => {
  const wrapper = shallow(generateComponent({
    data: {
      content: null,
    },
  }));
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it("parses series query", () => {
  expect(SERIES_QUERY).toMatchSnapshot();
});

it("updates nav on mount", () => {
  const mockDispatch = jest.fn();
  navActions.setLevel = jest.fn();
  const wrapper = shallow(generateComponent({
    dispatch: mockDispatch,
  }));
  expect(mockDispatch).toHaveBeenCalledTimes(1);
  expect(navActions.setLevel).toHaveBeenCalledTimes(1);
  expect(navActions.setLevel).toHaveBeenCalledWith("TOP");
});

it("handleRefresh calls refetch", () => {
  const mockRefetch = jest.fn(() => new Promise(p => p()));
  const wrapper = shallow(generateComponent({
    data: {
      refetch: mockRefetch,
    },
  }));
  wrapper.instance().handleRefresh();
  expect(mockRefetch).toHaveBeenCalledTimes(1);
});
