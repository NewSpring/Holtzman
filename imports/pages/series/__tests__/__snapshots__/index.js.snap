exports[`test parses series query 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "getSeries",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "channel",
                },
                "value": Object {
                  "kind": "StringValue",
                  "value": "series_newspring",
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "limit",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "limit",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "skip",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "skip",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "content",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": Object {
                    "kind": "Name",
                    "value": "entryId",
                  },
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "title",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "status",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "channelName",
                  },
                  "selectionSet": null,
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "meta",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "urlTitle",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "siteId",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "date",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "channelId",
                        },
                        "selectionSet": null,
                      },
                    ],
                  },
                },
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "content",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "sizes",
                            },
                            "value": Object {
                              "kind": "ListValue",
                              "values": Array [
                                Object {
                                  "kind": "StringValue",
                                  "value": "large",
                                },
                              ],
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "images",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "fileName",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "fileType",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "fileLabel",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "url",
                              },
                              "selectionSet": null,
                            },
                          ],
                        },
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "isLight",
                        },
                        "selectionSet": null,
                      },
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "colors",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "value",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "description",
                              },
                              "selectionSet": null,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "limit",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "skip",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
}
`;

exports[`test renders loading 1`] = `
<Connect(ApollosPullToRefresh)
  handleRefresh={[Function]}>
  <Meta
    title="Series" />
  <div
    className="background--light-secondary">
    <section
      className="soft-half">
      <div
        className="grid">
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItemSkeleton />
        </div>
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItemSkeleton />
        </div>
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItemSkeleton />
        </div>
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItemSkeleton />
        </div>
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItemSkeleton />
        </div>
        <div
          className="grid__item one-whole">
          <mockConstructor />
        </div>
      </div>
    </section>
  </div>
</Connect(ApollosPullToRefresh)>
`;

exports[`test renders with props 1`] = `
<Connect(ApollosPullToRefresh)
  handleRefresh={[Function]}>
  <Meta
    title="Series" />
  <div
    className="background--light-secondary">
    <section
      className="soft-half">
      <div
        className="grid">
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItem
            item={Object {}} />
        </div>
        <div
          className="grid__item one-half@palm-wide one-third@portable one-quarter@anchored flush-bottom@handheld push-bottom@portable push-bottom@anchored">
          <FeedItem
            item={Object {}} />
        </div>
        <div
          className="grid__item one-whole">
          <mockConstructor />
        </div>
      </div>
    </section>
  </div>
</Connect(ApollosPullToRefresh)>
`;
